sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/github.yaml

```diff
@@ -0,0 +1,78 @@
+provider:
+  metadata:
+    name: GitHub
+    category: Code hosting
+    message: Make sure this Github token gets revoked, changed, and removed from the code.
+  detection:
+    post:
+      heuristicFilter:
+        heuristics: # Not sure this is implemented yet. Will make sense when it is.
+          - path
+          - uri
+
+  rules:
+    - id: github-v2-tokens
+      metadata:
+        name: Github tokens should not be disclosed
+      detection:
+        matching:
+          pattern: "\\bgh[pusor]_(?<rand>[a-zA-Z0-9_]{36,255})\\b"
+        post:
+          statisticalFilter:
+            threshold: 4 # Based on significant sampling, lowest observed entropy is 4.1
+            inputString: $rand # Not sure this is implemented yet. Threshold should be OK with the full candidate
+      examples:
+        - text: >
+            gh_token=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
+          containsSecret: true
+          match: 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
+    
+    - id: github-fine-grained-tokens
+      metadata:
+        name: Github tokens should not be disclosed
+      detection:
+        matching:
+          pattern: "\\bgithub_pat_(?<rand>[a-zA-Z0-9]{22}_[a-zA-Z0-9]{59})\\b"
+        post:
+          statisticalFilter:
+            threshold: 4.8 # Based on significant sampling, 4.9 < entropy < 5.6
+            inputString: $rand # Not sure this is implemented yet. Threshold should be OK with the full candidate
+      examples:
+        - text: >
+            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
+          containsSecret: true
+          match: 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
+        - text: >
+            TOKEN=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
+          containsSecret: true
+          match: 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
+    
+    - id: github-oauth-access-tokens
+      metadata:
+        name: Github tokens should not be disclosed
+      detection:
+        matching:
+          pattern: "(?is)(client.?secret|access.?token).{1,30}?\\b(?<rand>[a-f0-9]{40})\\b"
+          context:
+            patternAround: "\\bIv1\\.[a-f0-9]{16}\\b"
```

Comment:
I can't see any obvious source for this pattern, and it doesn't appear in the GitGuardian documentation. Is it something you've seen when using SourceGraph?

Reply:
I did try to see if this was documented anywhere, but it's not. I do see examples dating back as far as November 2020, so the `Iv1.` prefix is at least that old.

If we wanted to detect older GitHub secrets, we'd need something like `(?i)github[^\/]{1,20}\b[0-9a-f]{20}\b` to match the old 20 hex char IDs. That could lead to too many FPs.